src/store/modules/util/actions.ts

```diff
@@ -1,33 +1,128 @@
+import { UtilService } from '@/services/UtilService'
 import { ActionTree } from 'vuex'
 import RootState from '@/store/RootState'
-import * as types from './mutation-types'
-import { hasError } from '@/utils'
 import UtilState from './UtilState'
-import { UtilService } from '@/services/UtilService'
+import * as types from './mutation-types'
+import { showToast } from '@/utils'
+import { hasError } from '@/adapter'
+import { translate } from '@hotwax/dxp-components'
 
 const actions: ActionTree<UtilState, RootState> = {
-  async getSecurityGroups({ commit, state }) {
-    // don't fetch security group information if we already have security groups available in state
-    // Added condition as security group information will not be changed frequently
-    if(state.securityGroups.length) {
-      return;
+  async getRoles({ commit }) {
+    let roles = []
```

Comment:
Add check to only fetch roles if the roles are not already present in the state.